src/mudlet.cpp

```diff
@@ -260,37 +260,17 @@ mudlet::mudlet()
     scanForQtTranslations(getMudletPath(qtTranslationsPath));
     loadTranslators(mInterfaceLanguage);
 
-    qDebug().nospace().noquote() << "mudlet::mudlet() INFO - styles available: \n" << QStyleFactory::keys().join(QChar::LineFeed);
+    qDebug().nospace().noquote() << "mudlet::mudlet() INFO - styles available: \n" << QStyleFactory::keys().join(QStringLiteral(", "));
     qDebug().nospace().noquote() << "                         style in use is: " << qApp->style()->objectName();
-    // In the following the replacement parameter(s) are supplied at points of use:
-#if defined(Q_OS_MACOS)
-    if (QStyleFactory::keys().contains(QLatin1String("macintosh"))) {
-        if (qApp->style()->objectName() == QLatin1String("macintosh", Qt::CaseInsensitive)) {
-            qDebug().nospace().noquote() << "mudlet::mudlet() INFO - \"macintosh\" has been detected as the style factory in use - QPushButton styling fix applied!";
-            mBG_ONLY_BUTTON_SSHEET = QStringLiteral("QPushButton {background-color: %1; border: 5px solid %1;}");
-            mFG_BG_BUTTON_SSHEET = QStringLiteral("QPushButton {color: %1; background-color: %2; border: 5px solid %2;}");
-        } else {
-            qDebug().nospace().noquote() << "mudlet::mudlet() INFO - \"macintosh\" has NOT been detected as the style factory in use!";
-            mBG_ONLY_BUTTON_SSHEET = QStringLiteral("QPushButton {background-color: %1;}");
-            mFG_BG_BUTTON_SSHEET = QStringLiteral("QPushButton {color: %1; background-color: %2;}");
-        }
-    }
-#elif defined(Q_OS_WIN32)
-    if (QStyleFactory::keys().contains(QLatin1String("windowsvista"))) {
-        if (qApp->style()->objectName() == QLatin1String("windowsvista", Qt::CaseInsensitive)) {
-            qDebug().nospace().noquote() << "mudlet::mudlet() INFO - \"windowsvista\" has been detected as the style factory in use - QPushButton styling fix applied!";
-            mBG_ONLY_BUTTON_SSHEET = QStringLiteral("QPushButton {background-color: %1; border: 1px solid #8f8f91;}");
-            mFG_BG_BUTTON_SSHEET = QStringLiteral("QPushButton {color: %1; background-color: %2; border: 1px solid #8f8f91;}");
-        } else {
-            qDebug().nospace().noquote() << "mudlet::mudlet() INFO - \"windowsvista\" has NOT been detected as the style factory in use!";
-            mBG_ONLY_BUTTON_SSHEET = QStringLiteral("QPushButton {background-color: %1;}");
-            mFG_BG_BUTTON_SSHEET = QStringLiteral("QPushButton {color: %1; background-color: %2;}");
-        }
+    if (QString stylefactory = qApp->style()->objectName(); QStringList{"windowsvista", "macintosh"}.contains(stylefactory)) {
```

Comment:
It is all very well trying to avoid repeating oneself but at the time I was NOT sure that the MacOs case had the same solution as the Windows one - you seem to have assume the same will work for both but last time I checked it didn't...

Reply:
My fixed commit did work, see https://github.com/Mudlet/Mudlet/pull/4846#issuecomment-787862593